src/ui/map.js

```diff
@@ -3410,7 +3414,7 @@ class Map extends Camera {
     _updateTerrain() {
         // Recalculate if enabled/disabled and calculate elevation cover. As camera is using elevation tiles before
         // render (and deferred update after zoom recalculation), this needs to be called when removing terrain source.
-        this.painter.updateTerrain(this.style, this.isMoving() || this.isRotating() || this.isZooming());
+        this.painter.updateTerrain(this.style, this._isDragging());
```

Comment:
Could we add a named variable so this last parameter is more explicit on the intent, e.g.:
```suggestion
        const adaptCameraAltitude = this._isDragging();
        this.painter.updateTerrain(this.style, adaptCameraAltitude);
```